McEliece doesn't compile with GCC 11

As identified in open-quantum-safe/liboqs#1035 and open-quantum-safe/liboqs#1036, Classic McEliece doesn't compile with GCC 11 due to mismatching function signatures:

```
FAILED: src/kem/classic_mceliece/CMakeFiles/classic_mceliece_348864_avx.dir/pqclean_mceliece348864_avx/fft_tr.c.o
/usr/bin/cc  -Iinclude -I../src/kem/classic_mceliece/pqclean_mceliece348864_avx -I../src/common/pqclean_shims -fPIC -fvisibility=hidden -march=native -Werror -Wall -Wextra -Wpedantic -Wstrict-prototypes -Wshadow -Wformat=2 -Wfloat-equal -Wwrite-strings -O3 -fomit-frame-pointer -fdata-sections -ffunction-sections -Wl,--gc-sections -mavx2 -mpopcnt -Wno-language-extension-token -std=gnu11 -MD -MT src/kem/classic_mceliece/CMakeFiles/classic_mceliece_348864_avx.dir/pqclean_mceliece348864_avx/fft_tr.c.o -MF src/kem/classic_mceliece/CMakeFiles/classic_mceliece_348864_avx.dir/pqclean_mceliece348864_avx/fft_tr.c.o.d -o src/kem/classic_mceliece/CMakeFiles/classic_mceliece_348864_avx.dir/pqclean_mceliece348864_avx/fft_tr.c.o -c ../src/kem/classic_mceliece/pqclean_mceliece348864_avx/fft_tr.c
../src/kem/classic_mceliece/pqclean_mceliece348864_avx/fft_tr.c:351:47: error: argument 1 of type ‘vec128[12]’ {aka ‘__m128i[12]’} with mismatched bound [-Werror=array-parameter=]
  351 | void PQCLEAN_MCELIECE348864_AVX_fft_tr(vec128 out[GFBITS], vec256 in[][ GFBITS ]) {
      |                                        ~~~~~~~^~~~~~~~~~~
In file included from ../src/kem/classic_mceliece/pqclean_mceliece348864_avx/fft_tr.c:6:
../src/kem/classic_mceliece/pqclean_mceliece348864_avx/fft_tr.h:11:40: note: previously declared as ‘vec128 *’ {aka ‘__m128i *’}
   11 | void PQCLEAN_MCELIECE348864_AVX_fft_tr(vec128 * /*out*/, vec256  /*in*/[][ GFBITS ]);
      |                                        ^~~~~~~~
```

--- a/crypto_kem/mceliece348864/META.yml
+++ b/crypto_kem/mceliece348864/META.yml
@@ -35,6 +35,7 @@ implementations:
             - Darwin
           required_flags:
             - sse4_1
+            - bmi1
             - popcnt
     - name: avx
       version: SUPERCOP-20191221

--- a/crypto_kem/mceliece348864/avx/bm.h
+++ b/crypto_kem/mceliece348864/avx/bm.h
@@ -8,7 +8,7 @@
 
 #include "vec128.h"
 
-void PQCLEAN_MCELIECE348864_AVX_bm(uint64_t * /*out*/, vec128 * /*in*/);
+void PQCLEAN_MCELIECE348864_AVX_bm(uint64_t /*out*/[ GFBITS ], vec128 /*in*/ [ GFBITS ]);
 
 #endif
 

--- a/crypto_kem/mceliece348864/avx/fft_tr.h
+++ b/crypto_kem/mceliece348864/avx/fft_tr.h
@@ -8,7 +8,7 @@
 #include "params.h"
 #include "vec256.h"
 
-void PQCLEAN_MCELIECE348864_AVX_fft_tr(vec128 [ GFBITS ] /*out*/, vec256  /*in*/[][ GFBITS ]);
+void PQCLEAN_MCELIECE348864_AVX_fft_tr(vec128 /*out*/[ GFBITS ], vec256  /*in*/[][ GFBITS ]);
 
 #endif
 

--- a/crypto_kem/mceliece348864/avx/fft_tr.h
+++ b/crypto_kem/mceliece348864/avx/fft_tr.h
@@ -8,7 +8,7 @@
 #include "params.h"
 #include "vec256.h"
 
-void PQCLEAN_MCELIECE348864_AVX_fft_tr(vec128 * /*out*/, vec256  /*in*/[][ GFBITS ]);
+void PQCLEAN_MCELIECE348864_AVX_fft_tr(vec128 [ GFBITS ] /*out*/, vec256  /*in*/[][ GFBITS ]);
 
 #endif
 

--- a/crypto_kem/mceliece348864/sse/bm.c
+++ b/crypto_kem/mceliece348864/sse/bm.c
@@ -130,7 +130,7 @@ static inline void get_coefs(gf *out, vec128 *in) {
 
 /* input: in, sequence of field elements */
 /* output: out, minimal polynomial of in */
-void PQCLEAN_MCELIECE348864_SSE_bm(uint64_t out[ GFBITS ], vec128 in[ GFBITS ]) {
+void PQCLEAN_MCELIECE348864_SSE_bm(uint64_t *out, vec128 *in) {
     uint16_t i;
     uint16_t N, L;
 

--- a/crypto_kem/mceliece348864f/avx/bm.c
+++ b/crypto_kem/mceliece348864f/avx/bm.c
@@ -129,7 +129,7 @@ static inline void get_coefs(gf *out, vec128 *in) {
 
 /* input: in, field elements in bitsliced form */
 /* output: out, field elements in non-bitsliced form */
-void PQCLEAN_MCELIECE348864F_AVX_bm(uint64_t out[ GFBITS ], vec128 in[ GFBITS ]) {
+void PQCLEAN_MCELIECE348864F_AVX_bm(uint64_t *out, vec128 *in) {
     uint16_t i;
     uint16_t N, L;
 

--- a/crypto_kem/mceliece348864f/avx/fft_tr.c
+++ b/crypto_kem/mceliece348864f/avx/fft_tr.c
@@ -348,7 +348,7 @@ static void butterflies_tr(vec128 out[ GFBITS ], vec256 in[][ GFBITS ]) {
     }
 }
 
-void PQCLEAN_MCELIECE348864F_AVX_fft_tr(vec128 out[GFBITS], vec256 in[][ GFBITS ]) {
+void PQCLEAN_MCELIECE348864F_AVX_fft_tr(vec128 *out, vec256 in[][ GFBITS ]) {
     butterflies_tr(out, in);
     radix_conversions_tr(out);
 }

--- a/crypto_kem/mceliece348864f/sse/bm.c
+++ b/crypto_kem/mceliece348864f/sse/bm.c
@@ -130,7 +130,7 @@ static inline void get_coefs(gf *out, vec128 *in) {
 
 /* input: in, sequence of field elements */
 /* output: out, minimal polynomial of in */
-void PQCLEAN_MCELIECE348864F_SSE_bm(uint64_t out[ GFBITS ], vec128 in[ GFBITS ]) {
+void PQCLEAN_MCELIECE348864F_SSE_bm(uint64_t *out, vec128 *in) {
     uint16_t i;
     uint16_t N, L;
 

